plugins/devtools-backend/src/service/router.ts

```diff
@@ -35,30 +35,68 @@ import { createPermissionIntegrationRouter } from '@backstage/plugin-permission-
 import {
   DiscoveryService,
   HttpAuthService,
+  LifecycleService,
+  LoggerService,
   PermissionsService,
 } from '@backstage/backend-plugin-api';
+import { SignalService } from '@backstage/plugin-signals-node';
 
 /** @public */
 export interface RouterOptions {
   devToolsBackendApi?: DevToolsBackendApi;
-  logger: Logger;
+  logger: LoggerService;
   config: Config;
   permissions: PermissionsService;
   discovery: DiscoveryService;
   httpAuth?: HttpAuthService;
+  lifecycle?: LifecycleService;
+  signalService?: SignalService;
 }
 
 /** @public */
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { logger, config, permissions } = options;
+  const { logger, config, permissions, lifecycle, signalService } = options;
 
   const { httpAuth } = createLegacyAuthAdapters(options);
 
   const devToolsBackendApi =
     options.devToolsBackendApi || new DevToolsBackendApi(logger, config);
 
+  const signalInterval =
+    config.getOptionalNumber('devtools.info.signalIntervalMs') ?? 1000;
+  if (lifecycle && signalService) {
+    let interval: NodeJS.Timeout | undefined;
+    lifecycle.addStartupHook(async () => {
+      interval = setInterval(async () => {
+        const info = await devToolsBackendApi.listInfo();
+        await signalService.publish<DevToolsInfo>({
+          channel: 'devtools:info',
```

Comment:
Just to double check here, this is intended to be efficient because only subscribers of the channel will receive the signal right? I.e. the total cost for this is essentially the sending of signals to the signal backend at `signalInterval`, and ofc anyone actually us subscribing to the signals but clearly that's something we'd pay for regardless of solution.

That's in contrast to the notifications where users are generally always subscribing to notifications, but we instead send much fewer signals to those and often narrow recipients to individual users.

Reply:
Yes, only connections subscribed to this signal will receive the payload